### PR TITLE
relised entity and CRUD operations

### PR DESCRIPTION
**Pull request by DiskReader**

### CODE REVIEW

File: CoffeeShop/Controllers/CoffeeController.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+﻿using CoffeeShop.Context;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace CoffeeShop.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CoffeeController : ControllerBase
```
Comment (hromau) [inline CoffeeShop/Controllers/CoffeeController.cs]: why u inherit ControllerBase?

File: CoffeeShop/Controllers/CoffeeController.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        [HttpGet(Name = "GetCoffee")]
+        public IEnumerable<Coffee> Get()
+        {
+            return db.Coffees;
+        }
+
+        [HttpPost]
+        public void Post([FromForm]Coffee coffee)
```
Comment (hromau) [inline CoffeeShop/Controllers/CoffeeController.cs]: [FromBody]

File: CoffeeShop/Context/CoffeeShopContext.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        public CoffeeShopContext()
+        {
+        }
+
+        public CoffeeShopContext(DbContextOptions<CoffeeShopContext> options)
+            : base(options)
+        {
+        }
+
+        public virtual DbSet<Coffee> Coffees { get; set; } = null!;
```
Comment (hromau) [inline CoffeeShop/Context/CoffeeShopContext.cs]: it looks strange. why u use here 'null!'?

File: CoffeeShop/Entities/Coffee.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+public class Coffee
+{
+    [Key]
+    public int Id { get; set; }
+
+	public string? Name { get; set; }
```
Comment (hromau) [inline CoffeeShop/Entities/Coffee.cs]: why the name is not required? i guess name should be required field. how u will order cofee without name?

**Review by CrazyKamenwik — APPROVED** [review]

File: CoffeeShop/Controllers/CoffeeController.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            db.Coffees.Add(coffee);
+            db.SaveChanges();
+        }
+
+        [HttpPut]
+        public void Put(int id, [FromForm] Coffee coffee)
```
Comment (hromau) [inline CoffeeShop/Controllers/CoffeeController.cs]: try to rename to more clearly names. for instance, GetById, DeleteById, GetList and etc

**Review by hromau — CHANGES_REQUESTED** [review]

**Review by pavelGagarin0870 — APPROVED** [review]

**Pull request merged**